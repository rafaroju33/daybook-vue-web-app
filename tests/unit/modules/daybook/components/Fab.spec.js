import Fab from "@/modules/daybook/components/Fab.vue";
import { shallowMount } from "@vue/test-utils";

describe("Testing the FAB component", () => {
  test("Should match to Snapshot", () => {
    const wrapper = shallowMount(Fab);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should contain the icon fa-plus by default", () => {
    const wrapper = shallowMount(Fab);
    const iTag = wrapper.find("i");
    expect(iTag.attributes("class")).toContain("fa-plus");
    expect(iTag.attributes("class")).toBeTruthy();
  });

  test("Show show the argument icon: fa-circle", () => {
    const wrapper = shallowMount(Fab, {
      props: {
        icon: "fa-circle",
      },
    });
    const iTag = wrapper.find("i");
    expect(iTag.attributes("class")).toContain("fa-circle");
    expect(iTag.classes("fa-circle")).toBeTruthy();
  });
  test("should emit the event on:click when click on it", () => {
    const wrapper = shallowMount(Fab);
    wrapper.find("button").trigger("click");
    expect(wrapper.emitted("on:click")).toHaveLength(1);
  });
});
