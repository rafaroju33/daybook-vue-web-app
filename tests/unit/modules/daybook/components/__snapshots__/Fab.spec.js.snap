// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the FAB component Should match to Snapshot 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
